### DISCUSSION TITLE
Where to store custom queries

### DISCUSSION THREAD

**Discussion by testhej** [HiPhish/rainbow-delimiters.nvim#114] [Q&A] [answered]
Hello, I have modified tsx/rainbow-parens.scm query to my liking, my question is where and or how to save this custom version so that it wont be reset when updating the plugin or cause any problems and how to load it in the config. Currently I load it like
            query = {
                tsx = "rainbow-parens",
            },

**Comment by HiPhish** [comment]
You should be able to save it under `after/queries/tsx/rainbow-parens.scm` in your configuration. Technically you can name the file whatever you want, but then you have to set the name of the query in your rainbow delimiters settings. Further reading:

- `:h 'runtimepath'`
- `:h treesitter`
- `:h tresitter-query`
- `:h rb-delimiters-query`
- `:h rb-delimiters-custom-query`

**Comment by testhej** [comment, marked as answer]
Hey! Thanks for the response, I was unable to make it work when calling it rainbow-parens.scm and placing it in after/queries/tsx/rainbow-parens.scm and then overriding in lua config like:
query = {
    tsx = "rainbow-parens",
},
However simply changing the name to something else like rainbow-custom did the trick!
